src/pages/workspace/categories/SpendCategorySelectorListItem.tsx

```diff
@@ -0,0 +1,61 @@
+import React, {useState} from 'react';
+import BaseListItem from '@components/SelectionList/BaseListItem';
+import type {BaseListItemProps, ListItem} from '@components/SelectionList/types';
+import useThemeStyles from '@hooks/useThemeStyles';
+import blurActiveElement from '@libs/Accessibility/blurActiveElement';
+import CategorySelector from '@pages/workspace/distanceRates/CategorySelector';
+import * as Policy from '@userActions/Policy/Policy';
+
+function SpendCategorySelectorListItem<TItem extends ListItem>({item, onSelectRow, isFocused}: BaseListItemProps<TItem>) {
+    const styles = useThemeStyles();
+    const [isCategoryPickerVisible, setIsCategoryPickerVisible] = useState(false);
+    const {policyID, groupID, categoryID} = item;
+
+    if (!policyID || !groupID) {
+        return;
+    }
+
+    const onSelect = (data: TItem) => {
+        setIsCategoryPickerVisible(true);
+        onSelectRow(data);
+    };
+
+    const setNewCategory = (selectedCategory: ListItem) => {
+        if (!selectedCategory.text) {
+            return;
+        }
+        Policy.setWorkspaceDefaultSpendCategory(policyID, groupID, selectedCategory.text);
+    };
```

Comment:
When setting a new category we had to check that it's not the current selected one i.e. newCategory != currentCategory. This is because we have a case where if you go offline, delete the current category and select it again the following API calls will be queued:
1. Delete category
2. Set spend category

And in the second API call we will be setting the spend category to a category that no longer exists which will causes an error.

(Comnig from https://github.com/Expensify/App/issues/49295)